haystack/utils/import_utils.py

```diff
@@ -62,6 +64,28 @@ def _optional_component_not_installed(component: str, dep_group: str, source_err
     ) from source_error
 
 
+def load_documents_from_remote(dataset_name: str, split: Optional[str] = "train") -> List[Document]:
+    """
+    Load a list of Haystack Documents from a remote Hugging Face dataset.
+
+    :param dataset_name: A Hugging Face dataset containing Haystack Documents
+    :param split: Optional parameter. The split of the Hugging Face dataset to load from. By default this is set to "train".
+    :return a List of Documents
+    """
+    try:
+        dataset = load_dataset_builder(dataset_name)
+        document_keys = set(["content", "content_type"])
+        if not document_keys.issubset(dataset.info.features.keys()):
+            raise Exception("This dataset does not conntain Haystack Documents.")
+    except FileNotFoundError as e:
```

Comment:
Let’s not safeguard this. Let this function raise an exception and instead of failing silently. The user should handle the exception.